src/main/java/gregtech/integration/baubles/BaubleBehavior.java

```diff
@@ -0,0 +1,52 @@
+package gregtech.integration.baubles;
+
+import baubles.api.BaubleType;
+import baubles.api.IBauble;
+import baubles.api.cap.BaublesCapabilities;
+import gregtech.api.items.metaitem.stats.IItemCapabilityProvider;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class BaubleBehavior implements IItemCapabilityProvider, ICapabilityProvider, IBauble {
+
+    private final BaubleType baubleType;
+
+    public BaubleBehavior(BaubleType baubleType) {
+        this.baubleType = baubleType;
+    }
+
+    @Override
+    public BaubleType getBaubleType(ItemStack stack) {
+        return baubleType;
+    }
+
+    @Override
+    public void onWornTick(ItemStack stack, EntityLivingBase player) {
+        if (stack != null && stack != ItemStack.EMPTY && player != null) {
+            stack.getItem().onUpdate(stack, player.getEntityWorld(), player, 0, false);
```

Comment:
Can this lead to potentially ticking items twice if baubles is installed?

Reply:
No because Item#onUpdate() is only called when in player inventory and shouldnt include baubles slots